test/integration/specter-wallet.spec.ts

```diff
@@ -0,0 +1,86 @@
+import { baseLogger } from "src/logger"
+import { UserWallet } from "src/userWallet"
+import { SpecterWallet } from "src/SpecterWallet"
+import { getActiveOnchainLnd } from "src/lndUtils"
+import { bitcoindClient, getChainBalance, mineBlockAndSyncAll } from "test/helpers"
+import { MainBook } from "src/mongodb"
+import { bitcoindAccountingPath } from "src/ledger/ledger"
+
+const { lnd } = getActiveOnchainLnd()
+const specterWalletName = "specter/coldstorage"
+let specterWallet
+
+beforeEach(async () => {
+  UserWallet.setCurrentPrice(10000)
+  specterWallet = new SpecterWallet({ logger: baseLogger })
+})
+
+afterAll(async () => {
+  await bitcoindClient.unloadWallet({ wallet_name: specterWalletName })
+})
+
+describe("SpecterWallet", () => {
+  it("creates wallet", async () => {
+    let wallets = await SpecterWallet.listWallets()
+
+    if (wallets.length < 2) {
+      try {
+        await SpecterWallet.createWallet()
+      } catch {
+        const { name } = await bitcoindClient.loadWallet(specterWalletName)
+        expect(name).toBe(specterWalletName)
+      }
+    }
+
+    wallets = await SpecterWallet.listWallets()
+    expect(wallets.length).toBe(2)
+  })
+
+  it("deposit to bitcoind", async () => {
+    const initBitcoindBalance = await specterWallet.getBitcoindBalance()
+    const { chain_balance: initLndBalance } = await getChainBalance({ lnd })
+
+    const sats = 10000
+    await specterWallet.toColdStorage({ sats })
+    await mineBlockAndSyncAll(/* 1001 */)
+
+    const bitcoindBalance = await specterWallet.getBitcoindBalance()
+    const { chain_balance: lndBalance } = await getChainBalance({ lnd })
+
+    expect(bitcoindBalance).toBe(initBitcoindBalance + sats)
+
+    const {
+      results: [{ fee }],
+    } = await MainBook.ledger({
+      account: bitcoindAccountingPath,
+      type: "to_cold_storage",
+    })
+
+    baseLogger.debug({
+      lndBalance,
+      initLndBalance,
+      sats,
+      fee,
+      bitcoindBalance,
+      initBitcoindBalance,
+    })
+    expect(lndBalance).toBe(initLndBalance - sats - fee)
+  })
+
+  // TODO: Fix or remove. Expectations were commented out
+  it.skip("withdraw from bitcoind", async () => {
```

Comment:
now that specter has it's API available, we should be able to do more integration/testing on that.

https://github.com/cryptoadvance/specter-desktop/pull/1232